lib/sentence-generator/index.js

```diff
@@ -249,10 +275,23 @@ class SentenceGenerator extends events.EventEmitter {
         for (let preprocessed of ex.preprocessed) {
             let grammarCat = 'thingpedia_' + ex.type;
 
+            if (grammarCat === 'thingpedia_query') {
+                const human_entity_types = ['tt:contact', 'tt:contact_name', 'org.wikidata:human'];
+                if (ex.value.schema.hasArgument('id')) {
+                    let type = ex.value.schema.getArgument('id').type;
+                    if (type.isEntity && human_entity_types.includes(type.type)) {
+                        grammarCat = 'thingpedia_who_question';
+                        this._grammar.addRule(grammarCat, [''], $runtime.simpleCombine(() => ex.value));
+                        continue;
+                    }
+                }
```

Comment:
What's going on here? This code should be outside the `for (let preprocessed of ex.preprocessed)` loop right? Otherwise you add multiple rules? And you're adding rules with empty string for expansion?

Also, `tt:contact_name` is obsolete, we should use `tt:username` consistently to mean unresolved phone_number/email_address. And `tt:contact` is a type used by Comma to mean email/phone/matrix account, which doesn't seem relevant here?
Do you expect there will be a `@builtin.contacts()` that uses the OS address book (or some thingengine-core address book abstraction) and has an `id` parameter of type `tt:contact`?

Reply:
Fixed. `tt:contact` is still included in case we might want to use it in the future as in the example you provided.